src/main.rs

```
@@ -35,33 +35,32 @@ fn main() {
     let pattern = matches.value_of("pattern").unwrap_or(&empty_pattern);
 
     // Get the current working directory
-    let current_dir_buf = match env::current_dir() {
-        Ok(cd) => cd,
-        Err(_) => error("Error: could not get current directory."),
-    };
-    let current_dir = current_dir_buf.as_path();
+    let current_dir = Path::new(".");
```

Why did you change this?

The detection of the current directory being missing does not seem to work anymore.

---

So it seems that .is_dir() doesn't really check the existence of path. :(

---

Maybe `Path::exists` helps? Does this have to be changed?

---

.exists() doesn't help too, on both stable&nightly branches. Path::new("") is ok, but "." and ".." are not.

---

Oh, "" doesn't mean current directory. I will ask the rust team.

---

What was wrong with the old way (`env::current_directory()`)?

---

env::current_dir() does check the existence. The returned path is resolved, so its value is used only when we don't have an absolute path.

---

`if let Err(_) = env::current_dir()` can help, but for root_dir, user still can pass in "." or "..". I hope there is an elegant way to check their existence.

---

Now I'm not sure if we should care the existence of "." and "..". Just let it crash?